ziti/contexts.go

```diff
@@ -98,9 +98,11 @@ func NewContextWithOpts(cfg *Config, options *Options) (Context, error) {
 		newContext.maxDefaultConnections = 1
 	}
 
-	idCredentials := edge_apis.NewIdentityCredentialsFromConfig(cfg.ID)
-	idCredentials.ConfigTypes = cfg.ConfigTypes
-	cfg.Credentials = idCredentials
+	if cfg.Credentials == nil {
+		idCredentials := edge_apis.NewIdentityCredentialsFromConfig(cfg.ID)
+		idCredentials.ConfigTypes = cfg.ConfigTypes
+		cfg.Credentials = idCredentials
+	}
```

Comment:
would you consider adding a warning in the else here? this might be confusing to people imo. Or return an error ?